### ISSUE TITLE
Warn on 'await function'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#413]
```js
await function foo() {
}
```
We should warn about the `await`. They probably meant `async`.

We shouldn't warn in these cases though:
```js
await function foo() {
}();
```
```js
await (function foo() {
});
```

**Comment by HicaroD** [comment]
Hi, Strager. Can you assign it to me? I don't expect payment for this task.

Actually, I already solved the problem, spent some time for playing around with it and got something working. Here is the PR in case you're interested: https://github.com/quick-lint/quick-lint-js/pull/1187
